src/peft/tuners/adalora/layer.py

```diff
@@ -95,64 +95,58 @@ def __init__(
         self.active_adapter = adapter_name
 
     def merge(self) -> None:
-        if self.active_adapter not in self.lora_A.keys():
-            return
         if self.merged:
-            warnings.warn("Already merged. Nothing to do.")
-            return
-        if self.r[self.active_adapter] > 0:
-            self.weight.data += (
-                transpose(
-                    self.lora_B[self.active_adapter]
-                    @ (self.lora_A[self.active_adapter] * self.lora_E[self.active_adapter]),
-                    self.fan_in_fan_out,
-                )
-                * self.scaling[self.active_adapter]
-                / (self.ranknum[self.active_adapter] + 1e-5)
+            warnings.warn(
+                f"Already following adapters were merged {','.join(self.merged_adapters)}. "
+                f"You are now additionally merging {','.join(self.active_adapters)}."
```

Comment:
Maybe print the warning only in the case `len(self.merged_adapters)>1`

Reply:
Perhaps that's not even needed as it seems you always merge all adapters

Reply:
Resolving this as discussed internally.